File: weather-station/csharp/src/WeatherStation/Statistics.cs
namespace WeatherStation;

public readonly record struct Statistics(
    decimal MinTemperature,
    decimal MaxTemperature,
    decimal AvgTemperature,
    decimal MinHumidity,
    decimal MaxHumidity,
    decimal AvgHumidity,
    decimal MaxWindSpeed,
    decimal AvgWindSpeed);

File: weather-station/typescript/src/Statistics.ts
export class Statistics {
  constructor(
    public readonly minTemperature: number,
    public readonly maxTemperature: number,
    public readonly avgTemperature: number,
    public readonly minHumidity: number,
    public readonly maxHumidity: number,
    public readonly avgHumidity: number,
    public readonly maxWindSpeed: number,
    public readonly avgWindSpeed: number,
  ) {}
}
